### ISSUE TITLE
paddlecloud GPU集群更新lasted image后出现cost一直不变（原集群收敛正常）

### ISSUE THREAD

**Issue by fanfannothing** [PaddlePaddle/Paddle#7120]
**【问题描述】**
dlnel gpu集群，上一个版本会出现event无gm成员变量，导致无法获取 layer的信息。因此，paddle cloud值班同学更新image（lasted版本）后，之前运行且收敛正常的程序，现在cost不变，一直为0.693147这个固定值（CPU集群为原始版本，收敛正常，但更新后的GPU集群出现上述错误）
我两位同事的任务（之前收敛正常），在这个集群启动任务，也出现上述同样的错误。

**【配置】**
- 1、任务提交方式

>   5 ../../cloud/bin/paddlecloud submit -jobname my-paddlecloud-job-xxxxv1 \
>   6   -cpu 1 \
>   7   -gpu 8 \
>   8   -memory 40Gi \
>   9   -parallelism 1 \
>  10   -pscpu 1 \
>  11   -pservers 1 \
>  12   -psmemory 1Gi \
>  13   -entry "sh run.sh" \
>  14   `pwd`/xxx

- 2、运行方式

>  18 python train.py --cloud=1 \
>  19                         -y 1 --model_arch 1 \
>  20                         --target_feature_len=2 \
>  21                         --batch_size=128 \
>  22                         --use_gpu=True \
>  23                         --num_passes=30 \

- 3、网络结构 （类dssm的结构）

> cnn（3种粒度）->fc层（256、128、64大小级联）-> sim -> rank loss

> 与paddle 同学沟通，初步定位是属于paddle程序问题。
此bug应该会导致大部分的任务都会出现这种情况，影响比较恶劣，麻烦尽快跟进。谢谢

**Comment by Yancey1989** [comment]
Hi, @fanfannothing 
1. 麻烦贴下网络配置吧，DLNEL集群之前镜像的版本比较老，不确定最新的code是否在API上有所变化。
1. 请问本地单机运行可以正常收敛吗？

**Comment by fanfannothing** [comment]
网络配置如下：
`105     def create_embedding(self, input, prefix=""):
106         """
107         Create word embedding. The `prefix` is added in front of the name of
108         embedding"s learnable parameter.
109         """
110         logger.info("Create embedding table [%s] whose dimention is %d. " %
111                     (prefix, 200))
112         emb = paddle.layer.embedding(
113             input=input,
114             size=200,
115             param_attr=ParamAttr(name="%s_emb.w" % prefix, initial_std=0, is_static=False))
116         return emb`

`174     def create_cnn(self, emb, prefix=""):
175         """
176         A multi-layer CNN.
177
178         :param emb: The word embedding.
179         :type emb: paddle.layer
180         :param prefix: The prefix will be added to of layers' names.
181         :type prefix: str
182         """
183
184         def create_conv(context_len, hidden_size, prefix):
185             key = "%s_%d_%d" % (prefix, context_len, hidden_size)
186             # set parameter attr for parameter sharing
187             conv = paddle.networks.sequence_conv_pool(
188                 input=emb,
189                 context_len=context_len,
190                 hidden_size=hidden_size,
191                 context_proj_param_attr=ParamAttr(name=key + "contex_proj.w"),
192                 fc_param_attr=ParamAttr(name=key + "_fc.w"),
193                 fc_bias_attr=ParamAttr(name=key + "_fc.b"),
194                 pool_bias_attr=ParamAttr(name=key + "_pool.b"))
195             return conv
196
197         logger.info("create a sequence_conv_pool which context width is 2, prefix is %s" %(prefix+"cnn"))
198         conv_2 = create_conv(2, self.dnn_dims[1], prefix+"cnn")
199         logger.info("create a sequence_conv_pool which context width is 3, prefix is %s" %(prefix+"cnn"))
200         conv_3 = create_conv(3, self.dnn_dims[1], prefix+"cnn")
201         logger.info("create a sequence_conv_pool which context width is 4, prefix is %s" %(prefix+"cnn"))
202         conv_4 = create_conv(4, self.dnn_dims[1], prefix+"cnn")
203
204         return paddle.layer.concat(input=[conv_2, conv_3, conv_4])`

`206     def create_dnn(self, sent_vec, prefix, is_stacked=True):
207         # if more than three layers, than a fc layer will be added.
208         if len(self.dnn_dims) > 1:
209             _input_layer = sent_vec
210             for id, dim in enumerate(self.dnn_dims[0:]): # TODO
211                 name = "%s_fc_%d_%d" % (prefix, id, dim)
212                 logger.info("create fc layer [%s] which dimention is %d" %
213                             (name, dim))
214                 fc = paddle.layer.fc(
215                     input=_input_layer,
216                     size=dim,
217                     act=paddle.activation.Tanh(),
218                     param_attr=ParamAttr(name="%s.w" % name),  # TODO
219                     bias_attr=ParamAttr(name="%s.b" % name, initial_std=0.))
220                 _input_layer = fc
221
222         # bn
223         #_input_layer = paddle.layer.batch_norm(
224         #        input=_input_layer,
225         #        act=paddle.activation.Relu(),
226         #        param_attr=ParamAttr(name="%s_bn_s.w" %(prefix)),
227         #        bias_attr=ParamAttr(name="%s_bn_s.b" %(prefix), initial_std=0.),
228         #        layer_attr=paddle.attr.Extra(drop_rate=0.5))
229         #return _input_layer
230
231         _dropout_input_layer = paddle.layer.dropout(
232                 input=_input_layer,
233                 dropout_rate=0.5)
234
235         return _dropout_input_layer`

`249     def _build_rank_model(self):
250         """
251         Build a pairwise rank model, and the cost is returned.
252
253         A pairwise rank model has 3 inputs:
254           - source sentence
255           - left_target sentence
256           - right_target sentence
257           - label, 1 if left_target should be sorted in front of
258                    right_target, otherwise 0.
259         """
260         logger.info("build rank model")
261         assert self.model_type.is_rank()
262
263         prefixs = "_ _ _".split(
264         ) if self.share_semantic_generator else "source target target".split()
265         embed_prefixs = "_ _".split(
266         ) if self.share_embed else "source target".split()
267
268         # Source
269         source = paddle.layer.data(
270             name="source_input",
271             type=paddle.data_type.integer_value_sequence(self.vocab_sizes[0]))
272         source_emb = self.create_embedding(source, prefix=embed_prefixs[0])
273         source_sem = self.model_arch_creater(source_emb, prefix=prefixs[0])
274
275         # Left
276         left_target_list = []
277         for i_ in range(self.target_feature_len):
278             x = paddle.layer.data(
279                 name="left_target_input_%d" %(i_),
280                 type=paddle.data_type.integer_value_sequence(self.vocab_sizes[1]))
281             logger.info("left prefix is %s" %(embed_prefixs[1]+str(i_)))
282             x = self.create_embedding(x, prefix=embed_prefixs[1]+str(i_))
283             x = self.model_arch_creater(x, prefix=prefixs[1]+str(i_))
284             left_target_list.append(x)
285         # Merge the multi left input.(Use fc)
286         left_combined_features = paddle.layer.fc(
287                 input=left_target_list,
288                 name='left_target_merge',
289                 size=self.dnn_dims[2], # TODO
290                 act=paddle.activation.Tanh(),
291                 param_attr=[ParamAttr(initial_std=0., name="%s_fc.w" %(prefixs[1])) for x in left_target_list],
292                 bias_attr=[ParamAttr(name="%s_fc.b" % prefixs[1], initial_std=0.) for x in left_target_list])
294         # Right
295         right_target_list = []
296         for i_ in range(self.target_feature_len):
297             x = paddle.layer.data(
298                 name="right_target_input_%d" %(i_),
299                 type=paddle.data_type.integer_value_sequence(self.vocab_sizes[1]))
300             logger.info("right prefix is %s" %(embed_prefixs[1]+str(i_)))
301             x = self.create_embedding(x, prefix=embed_prefixs[1]+str(i_))
302             x = self.model_arch_creater(x, prefix=prefixs[2]+str(i_))
303             right_target_list.append(x)
304         # Merge the multi right input.(Use fc)
305         right_combined_features = paddle.layer.fc(
306                 input=right_target_list,
307                 name='right_target_merge',
308                 size=self.dnn_dims[2], # TODO
309                 act=paddle.activation.Tanh(),
310                 param_attr=[ParamAttr(initial_std=0., name="%s_fc.w" %(prefixs[2])) for x in right_target_list],
311                 bias_attr=[ParamAttr(name="%s_fc.b" % prefixs[2], initial_std=0.) for x in right_target_list])
312
313         if not self.is_infer:
314             label = paddle.layer.data(
315                 name="label_input", type=paddle.data_type.integer_value(1))
316
317         def sim_fc(input_layer, name, prefix_=""):
318             dnn = paddle.layer.fc(
319                     input=input_layer,
320                     name=name,
321                     size=1,
322                     act=paddle.activation.Linear(),
323                     param_attr=[ParamAttr(initial_std=0., name="%s_fc.w" %(prefix_)) for x in input_layer],
324                     bias_attr=[ParamAttr(name="%s_fc.b" % prefix_, initial_std=0.) for x in input_layer])
325             return dnn
326
327         # map fc to a score as a sim metric (It's GOOD)
328         prefix_share = 'score_share'
329         left_score = sim_fc([source_sem, left_combined_features], "left_score", prefix_=prefix_share)
330         right_score = sim_fc([source_sem, right_combined_features], "right_score", prefix_=prefix_share)

344         if not self.is_infer:
345             # rank cost
346             cost = paddle.layer.rank_cost(left=left_score, right=right_score, label=label)
347             return cost, None, label
348
349         return left_score, right_score
`

**Comment by fanfannothing** [comment]
@qingqing01 麻烦帮忙看看！谢谢了

**Comment by typhoonzero** [comment]
Close due to low activity, feel free to reopen.
